### PR TITLE
Remove sending phone number if not present

### PR DESCRIPTION
**Pull request by tarunkumar2**
The application is sending the phone number as empty value if the phone number is added by the user. Have fixed the application to prevent sending the phone number if the user doesn't enter the phone number.

### CODE REVIEW

**Comment by MasterXen** [comment]
All these changes can be simplified since empty string are falsy. Alternately you could use Typescript's [nullish coalescing functionality](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing). So this:
```ts
phoneNumber: this.formData.phoneNumber !== '' ? this.formData.phoneNumber : undefined
``` 
can become
```ts
phoneNumber: this.formData.phoneNumber || undefined
```

Do you have an easy way to test the API input being generated locally?

**Comment by tarunkumar2** [comment]
No @MasterXen . I am currently using the chrome inspect tool to test the API calls and the params sent.

**Comment by kristinfritsch** [comment]
@MasterXen @tarunkumar2 I’d prefer to have that logic on api layer, so we can actually write tests for it and the views don't need care about it.

**Comment by timmy-circle** [comment]
Nothing related to this PR. But I have a little suggestion on the UI/UX part for phone number. While testing on sample app, I had issue with not having proper format of my phone number that failed the api request. One little suggestion would be maybe have two separate box on phone number, one for country code and one for actual phone number. Wdyt? 
68747470733a2f2f7261772e6769746875622e636f6d2f6a61636b6f636e722f696e746c2d74656c2d696e7075742f6d61737465722f73637265656e73686f74732f76616e696c6c612e706e67

**Comment by tarunkumar2** [comment]
Thank you for the suggestion @timmy-circle . Yeah providing this kind of UI is better since it can avoid the API due to the improper phone number format. I will try to implement this kind of UI option along or provide the placeholder that will help the user determining the format.

File: lib/cardsApi.ts [hunk 1]
```diff
@@ -71,6 +71,22 @@ instance.interceptors.response.use(
   }
 )
 
+function checkNullInMetaData(metaData: MetaData) {
+  const nullIfEmpty = (prop: string | undefined) => {
+    if (prop === '') {
```
Comment (kristinfritsch) [inline lib/cardsApi.ts:76]: Probably a good idea to do `prop.trim() === ''` to also cover whitespace in a string

File: lib/cardsApi.ts [hunk 2]
```diff
@@ -71,6 +71,22 @@ instance.interceptors.response.use(
...
+    }
+    return prop
+  }
+  const updateMetaObj = {
+    email: nullIfEmpty(metaData.email),
+    phoneNumber: nullIfEmpty(metaData.phoneNumber),
+    sessionId: metaData.sessionId,
+    ipAddress: metaData.ipAddress
+  }
+  return updateMetaObj
```
Comment (kristinfritsch) [inline lib/cardsApi.ts:87]: This replaces the actual `metaData` object. If there are new properties added, they will be stripped out without noticing.

Best here is to copy the existing obj and only overwrite the properties we want to change.

```
const updateMetaObj = {
    ...metaData
    email: nullIfEmpty(metaData.email),
    phoneNumber: nullIfEmpty(metaData.phoneNumber),
  }
```

File: lib/cardsApi.ts [hunk 3]
```diff
@@ -71,6 +71,22 @@ instance.interceptors.response.use(
   }
 )
 
+function checkNullInMetaData(metaData: MetaData) {
+  const nullIfEmpty = (prop: string | undefined) => {
```
Comment (kristinfritsch) [inline lib/cardsApi.ts:75]: I'd like to have `nullIfEmpty` reusable across this file. Can we move it out of this function please.

File: lib/cardsApi.ts [hunk 4]
```diff
@@ -137,9 +138,9 @@ function getCards(pageBefore: string, pageAfter: string, pageSize: string) {
  */
 function createCard(payload: CreateCardPayload) {
   const url = `/v1/cards`
-  payload.metadata = checkNullInMetaData(payload.metadata)
-
-  return instance.post(url, payload)
+  const modifiedPayload = Object.assign({}, payload)
```
Comment (kristinfritsch) [inline lib/cardsApi.ts:141]: I don't think this fixes the issue. Try to use payload:
```
 {
    email: 'my-email@xyz.com',
    phoneNumber: '+491777777',
    sessionId: 'xyz',
    ipAddress: '10.0.2.1',
    newProp: 'my-prop'
  }
```

`newProp` will be omitted. That's why I suggested the spread syntax.

Comment (tarunkumar2) [reply lib/cardsApi.ts]: Sorry for not understanding the previous comment. I have updated the code now

File: lib/cardsApi.ts [hunk 5]
```diff
@@ -133,8 +141,9 @@ function createCard(payload: CreateCardPayload) {
  */
 function updateCard(cardId: string, payload: UpdateCardPayload) {
   const url = `/v1/cards/${cardId}`
-
-  return instance.put(url, payload)
+  payload.metadata.email = nullIfEmpty(payload.metadata.email)
+  payload.metadata.phoneNumber = nullIfEmpty(payload.metadata.phoneNumber)
+  return instance.post(url, payload)
```
Comment (kristinfritsch) [inline lib/cardsApi.ts:146]: why did that change to post instead of put?

File: lib/cardsApi.ts [hunk 6]
```diff
@@ -133,7 +141,8 @@ function createCard(payload: CreateCardPayload) {
  */
 function updateCard(cardId: string, payload: UpdateCardPayload) {
   const url = `/v1/cards/${cardId}`
-
+  payload.metadata.email = nullIfEmpty(payload.metadata.email)
```
Comment (kristinfritsch) [inline lib/cardsApi.ts:144]: Can we just have a small check if `metadata` prop is not undefined. Otherwise i think looks good

**Pull request merged**